### PR TITLE
update

### PR DESCRIPTION
**Pull request by aihao2000**
# What does this PR do?
- load_dataset data_dir
   Supports custom local datasets. Just like https://github.com/huggingface/diffusers/pull/5747#event-10922633415
- Update fingerprint hash parameters. Avoid irrelevant parameters causing repeated calculations in the map cache

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@sayakpaul and @patrickvonplaten

### CODE REVIEW

**Comment by sayakpaul** [comment]
Thanks for your contributions! Can we please break this PR into multiple PRs as it targets multiple mutually exclusive areas?

**Comment by aihao2000** [comment]
@sayakpaul Yes, feel free to change it.

**Comment by sayakpaul** [comment]
Sorry for not being clear. I was asking you to tackle this in multiple PRs

**Comment by aihao2000** [comment]
@sayakpaul I'm not familiar with git and it seems a little troublesome. Are there any doubtful changes? I can restore that part.

**Comment by sayakpaul** [comment]
@aihao2000 a gentle ping here

**Comment by aihao2000** [comment]
@sayakpaul IIt is now only about the addition of the data_dir parameter. Is this okay?

File: examples/text_to_image/train_text_to_image_sdxl.py [hunk 1]
```diff
@@ -474,8 +474,6 @@ def parse_args(input_args=None):
         args.local_rank = env_local_rank
 
     # Sanity checks
-    if args.dataset_name is None and args.train_data_dir is None:
-        raise ValueError("Need either a dataset name or a training folder.")
```
Comment (lhoestq) [inline examples/text_to_image/train_text_to_image_sdxl.py:478]: maybe keep this one ?

**Review by lhoestq — APPROVED** [review]
Cool :) ust one comment:

File: examples/text_to_image/train_text_to_image_sdxl.py (hunk 1 above)
Comment (aihao2000) [reply examples/text_to_image/train_text_to_image_sdxl.py:478]: @lhoestq oh my bad

**Comment by yiyixuxu** [comment]
@aihao2000 can we address the last comment and get this merged?

**Comment by aihao2000** [comment]
@yiyixuxu Should I submit a new commit to restore this code?

**Comment by yiyixuxu** [comment]
@aihao2000 yes

**Comment by yiyixuxu** [comment]
can you run `make style`?

**Comment by yiyixuxu** [comment]
thanks @hlky !

**Pull request merged**